src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -291,17 +292,27 @@ class TooltipHTMLContent {
         (el as any).domBelongToZr = true;
         this.el = el;
         const zr = this._zr = api.getZr();
-        const appendToBody = this._appendToBody = opt && opt.appendToBody;
 
-        makeStyleCoord(this._styleCoord, zr, appendToBody, api.getWidth() / 2, api.getHeight() / 2);
-
-        if (appendToBody) {
-            document.body.appendChild(el);
+        let container: HTMLElement | null = null;
+        if (opt && opt.appendTo) {
+	    if(typeof opt.appendTo === 'string') {
+	    	container = document.querySelector(opt.appendTo)
+	    } else if (typeof opt.appendTo === 'function') {
+	    	container = opt.appendTo(api.getDom())
+	    } else if (opt.appendTo instanceof HTMLElement) {
+                container = opt.appendTo
+	    }
         }
```

Comment:
Can be simplified as

```ts
const appendTo = opt.appendTo;
const container = (
    isString(appendTo)
        ? document.querySelector(appendTo)
        : isDom(appendTo)
            ? appendTo
            : isFunction(appendTo) && appendTo()
) || api.getDom();
```

Reply:
Thanks. I am not aware of the functions like `isDom` and  `isFunction`.

But i have some different idea about the last part `|| api.getDom()`. It seem to have different logic to calculate the tooltip position in `makeStyleCoord`. Perviously, it was desided by `appendToBody `. it is hard to choose the correct calculation logic if we have that fallback.